### ISSUE TITLE
PR preprocess test should compare with base branch

### ISSUE THREAD

**Issue by lsf37** [seL4/seL4#1299]
Currently the preprocess test is looking at the AST diff between the verification manifest version of seL4 and the PR version of the code.

When the verification manifest is in sync with the seL4 master branch (which it is most of the time), this happens to be mostly what we want.

For the preprocess check in deployment it is in fact exactly what we want, but for PR checks it would make more sense to compare to the base of the PR than the current verification manifest:

- if the verification manifest is behind, we will be seeing the unrelated diff from how far it is behind on master plus the diff of the PR. That is not helpful to judge what the PR itself will change. 
- the base of the PR might not be the master branch, but a different PR. We are currently again seeing both diffs instead of just what the current pull request does.

**Comment by Indanz** [comment]
Has this been fixed yet?

**Comment by lsf37** [comment]
No, this is still happening (it's why we see preprocess failures on PRs when we haven't update the verification manifest yet, either when we forget on when there is an actual problem).
